nfc/libs/NfcCoreLib/lib/Nci/phNciNfc_Init.c

```diff
@@ -69,12 +92,140 @@ static NFCSTATUS phNciNfc_Init(void *pContext)
+    if (NULL != pNciContext)
+    {
+        if ((pNciContext->ResetInfo.NciVer & PH_NCINFC_VERSION_MAJOR_MASK) <=
+            (PH_NCI1xNFC_VERSION & PH_NCINFC_VERSION_MAJOR_MASK))
+            wStatus = phNciNfc_ProcessInitRspNci1x(pContext, Status);
+        else if ((pNciContext->ResetInfo.NciVer & PH_NCINFC_VERSION_MAJOR_MASK) ==
+            (PH_NCI2xNFC_VERSION & PH_NCINFC_VERSION_MAJOR_MASK))
+            wStatus = phNciNfc_ProcessInitRspNci2x(pContext, Status);
+    }
+    PH_LOG_NCI_FUNC_EXIT();
+    return wStatus;
+}
+
+static NFCSTATUS phNciNfc_ProcessInitRspNci2x(void *pContext, NFCSTATUS Status)
+{
+    NFCSTATUS wStatus = Status;
     phNciNfc_sInitRspParams_t *pInitRsp = NULL;
     pphNciNfc_Context_t pNciContext = pContext;
     uint8_t Offset = 0;
     uint16_t wTemp = 0;
 
     PH_LOG_NCI_FUNC_ENTRY();
+    if (NULL != pNciContext)
+    {
+        pInitRsp = &(pNciContext->InitRspParams);
+        if (pNciContext->RspBuffInfo.wLen >= 19)
+        {
+            if (pNciContext->RspBuffInfo.pBuff[Offset++] == PH_NCINFC_STATUS_OK)
+            {
+                wStatus = NFCSTATUS_SUCCESS;
+
+                /*NFCC Features*/
+                pInitRsp->NfccFeatures.DiscConfSuprt = pNciContext->RspBuffInfo.pBuff[Offset++];
+                pInitRsp->NfccFeatures.RoutingType = pNciContext->RspBuffInfo.pBuff[Offset++];
+                pInitRsp->NfccFeatures.PwrOffState = pNciContext->RspBuffInfo.pBuff[Offset++];
+                pInitRsp->NfccFeatures.Byte3 = pNciContext->RspBuffInfo.pBuff[Offset++];
+
+                /*Max No of Logical Connection supported*/
+                pInitRsp->MaxLogicalCon = pNciContext->RspBuffInfo.pBuff[Offset++];
+
+                /*Routing Table Size*/
+                pInitRsp->RoutingTableSize = pNciContext->RspBuffInfo.pBuff[Offset++]; /*LSB*/
+                wTemp = pNciContext->RspBuffInfo.pBuff[Offset++];
+                pInitRsp->RoutingTableSize = (pInitRsp->RoutingTableSize) | (wTemp << 8);
+
+                /*Control Packet Payload Length*/
+                pInitRsp->CntrlPktPayloadLen = pNciContext->RspBuffInfo.pBuff[Offset++];
+
+                /*Data Packet Payload Length of the static HCI connection*/
+                pInitRsp->DataHCIPktPayloadLen = pNciContext->RspBuffInfo.pBuff[Offset++];
+
+                /*Number of Credits of the Static HCI Connection*/
+                pInitRsp->DataHCINumCredits = pNciContext->RspBuffInfo.pBuff[Offset++];
+
+                /*Max NFC-V RF Frame Size*/
+                pInitRsp->MaxNFCVFrameSize = pNciContext->RspBuffInfo.pBuff[Offset++];
+                wTemp = (uint16_t)pNciContext->RspBuffInfo.pBuff[Offset++];
+                pInitRsp->MaxNFCVFrameSize = (pInitRsp->MaxNFCVFrameSize) | (wTemp << 8);
+
+                /*number of supported RF interfaces*/
+                pInitRsp->NoOfRfIfSuprt = pNciContext->RspBuffInfo.pBuff[Offset++];
+
+                if (pInitRsp->NoOfRfIfSuprt <= PH_NCINFC_CORE_MAX_SUP_RF_INTFS)
```

Comment:
I guess you mean we need to have another check making sure data + pInitRsp->NoOfRfIfSuprt  == RspBuffInfo.wLen for NCI 1x and NCI2x...

Going back to the NCI2.0 spec, I believe I did a mistake because the RF Interfaces description is not is the same format. There is now a TLV (Interface, Number of Extensions, Extensions list).